src/Definition/Reflection/ClassDefinition.php

```diff
@@ -16,14 +15,14 @@ class ClassDefinition implements ClassDefinitionInterface
     /** @var ReflectionClass */
     private $reflection;
 
-    /** @var Parameter[] */
+    /** @var Parameter[]|null */
     private $parameters;
 
-    /** @var string[] */
+    /** @var list<string>|null */
```

Comment:
if setting an empty array as default, we could avoid having `null` here imho

Reply:
Probably yes, but this would require additional refactoring to this class since `null` is used by getter methods. I just added the types to reflect the existing behaviour.